src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -114,6 +114,11 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             _socketType = socketType;
             _protocolType = protocolType;
 
+            if (addressFamily == AddressFamily.InterNetworkV6)
+            {
+                DualMode = false;
```

Comment:
This shouldn't be necessary. If for some reason it is for *NIX, we should move it within the PAL. 
As written, the above ctor will call `setsockopt` twice with conflicting values: was that the intent?